template/src/pdk.ts.ejs

```diff
@@ -25,11 +25,11 @@ export class <%- schema.name %> {
       ...obj,
       <% schema.properties.forEach(p => { -%>
         <% if (isDateTime(p)) { -%>
-          <%- p.name -%>: obj.<%- p.name -%> ? new Date(obj.<%- p.name -%>) : null,
+          <%- p.name -%>: obj.<%- p.name -%> == null ? obj.<%- p.name %> : new Date(obj.<%- p.name -%>),
```

Comment:
I started to suggest this and then went to clarify intent on another comms channel :smile:

```suggestion
          <%- p.name -%>: obj.<%- p.name -%> === null || obj.<%- p.name -%> === undefined ? obj.<%- p.name %> : new Date(obj.<%- p.name -%>),
```

The other bit buried in here is `===` instead of `==`, since the latter performs coercions. (so `undefined == null` is true but `undefined === null` is false.)